### PR TITLE
Added logic to create a safe hash object

### LINKED ISSUE #16: Using {} as a hash
I'm sure async is vulnerable to this as well, but in this example

```javascript
Insync.auto({
    'hasOwnProperty': function (callback) {
        callback(null, 'hasOwnProperty');
    },
    'bind' : function (callback) {
        setTimeout(function() {

            callback(null, 'bind');
        }, 100);
    }
}, function (err, results) {

    expect(err).to.not.exist();
    expect(results).to.deep.equal({
        hasOwnProperty: 'hasOwnProperty',
        bind: 'bind'
    });

    done();
});
```

will cause an obvious hard error. I suggest we create any collector objects with `var results = Object.create(null)`. While there will probably be a _slight_ performance hit for it, it will require the least amount of code rework.

This could be another place for a sweet macro by the way.

### PR DESCRIPTION
**Pull request by arb**
Closes #16

We should update this to use the native [`Object.setPrototypeOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf) at some point, but I was unable to figure out how to pollyfill this conditionally and hit 100% coverage.

As far as performance, this is what "bench" says in node 0.10.34:

>
Winner: proto
Compared with next highest (Object.create), it's:
29.46% faster
1.42 times as fast
0.15 order(s) of magnitude faster
A LITTLE FASTER

### CODE REVIEW

File: lib/util.js [hunk 1]
```diff
@@ -109,6 +109,16 @@ module.exports.log = internals.consoleFn('log');
 module.exports.dir = internals.consoleFn('dir');
 
 
+module.exports.hash = function (values) {
+
+    var result = values || {};
```
Comment (geek) [inline lib/util.js]: not seeing this function called with values.

File: lib/util.js [hunk 2]
```diff
@@ -109,6 +109,16 @@ module.exports.log = internals.consoleFn('log');
...
+    /*eslint-disable */
+    result.__proto__ = null;;
```
Comment (geek) [inline lib/util.js]: double `;`

File: lib/util.js [hunk 3]
```diff
@@ -109,6 +109,16 @@ module.exports.log = internals.consoleFn('log');
 module.exports.dir = internals.consoleFn('dir');
 
 
+module.exports.hash = function (values) {
```
Comment (cjihrig) [inline lib/util.js]: This function is also destructive towards `values` if one is passed in.

Comment (geek) [inline lib/util.js]: My point is that this is an internal utility function that only gets used one way... so you can simplify it.

**Comment by arb** [comment]
Updated `_hash` function so it's not destructive and does not access `__proto__` directly.

**Comment by arb** [comment]
Closing for now so we can get this tested in a less annoying way.

**Pull request closed without merging**